common/src/main/java/com/microsoft/identity/common/internal/platform/SecretKeyAccessor.java

```diff
@@ -0,0 +1,195 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.platform;
+
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+
+import com.microsoft.identity.common.exception.ClientException;
+
+import java.nio.charset.Charset;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.Certificate;
+import java.util.Arrays;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.Mac;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.experimental.Accessors;
+
+import static com.microsoft.identity.common.exception.ClientException.BAD_PADDING;
+import static com.microsoft.identity.common.exception.ClientException.INVALID_ALG_PARAMETER;
+import static com.microsoft.identity.common.exception.ClientException.INVALID_BLOCK_SIZE;
+import static com.microsoft.identity.common.exception.ClientException.INVALID_KEY;
+import static com.microsoft.identity.common.exception.ClientException.INVALID_PROTECTION_PARAMS;
+import static com.microsoft.identity.common.exception.ClientException.KEYSTORE_NOT_INITIALIZED;
+import static com.microsoft.identity.common.exception.ClientException.NO_SUCH_ALGORITHM;
+import static com.microsoft.identity.common.exception.ClientException.NO_SUCH_PADDING;
+
+@Builder
+@Accessors(prefix = "m")
+public class SecretKeyAccessor implements KeyAccessor {
+    private static final Charset UTF8 = Charset.forName("UTF-8");
+    private final DeviceKeyManager<KeyStore.SecretKeyEntry> mKeyManager;
+    private final CryptoSuite suite;
+    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
+    @Override
+    public byte[] encrypt(@NonNull final byte[] plaintext) throws ClientException {
+        final String errCode;
+        final Exception exception;
+        try {
```

Comment:
Vars in this scope could be `final`, rather than leave separate suggestions for each line I'll just leave a comment here

Reply:
should be addressed.